### ISSUE TITLE
osm dashboard not an option to share track to

### ISSUE THREAD

**Issue by nutpantz** [OpenTracksApp/OSMDashboard#143]
Using open tracks 3.25 and osm dashboard 2.13 from fdroid on a pixel 6 with Android 12 feb update. There is no option to pick open dashboard to view my track (after picking the map icon i in the upper right screen of you rem tracks). I have the options of open tracks, gpstest, and osmand (which will show my track from open tracks). There is no option to view it in osm dashboard.

I remember picking gpx the first time i tried to view the map but it never asked me what format again.

I expect this is an issue with osm dashboard not being considered a receiver of this data type since i can view the data in osmmaps.

**Comment by pstorch** [comment]
Hi @nutpantz are you sure you use the "map" icon? This sounds more like the "share" icon. Because if OpenTracks offers you gps statues status and osmand it seems to be the track share function. 
@dennisguse @rgmf any other idea?

**Comment by nutpantz** [comment]
That's what i thought i did the first time share not map.. but I've tried it about 40 times since. Uninstalling both apps reinstalling and trying again

**Comment by nutpantz** [comment]
Share gives me the option of open tracks and my email apps. Maps gives me open tracks, gps status and osmand

**Comment by pstorch** [comment]
Very strange. 
And the share button shows the same apps as option?

**Comment by nutpantz** [comment]
No share button shows open  tracks and my 3 email apps.

**Comment by nutpantz** [comment]
![Screenshot_20220304-134032_Settings](https://user-images.githubusercontent.com/12160507/156845547-dacc5a3b-e7f5-4346-b1a5-183a6f8df54e.png)


This might help

**Comment by nutpantz** [comment]
![Screenshot_20220304-134349_Pixel Launcher](https://user-images.githubusercontent.com/12160507/156845895-a529a9dc-d608-4889-8b51-414cdb650cf7.png)

**Comment by rgmf** [comment]
> @dennisguse @rgmf any other idea?

I don't know, it's strange because I can do it using my Pixel 4a with Android 12 (with OpenTracks 3.25 and OSM Dashboar 2.13 :thinking: )

Maybe it's related but I don't have osmand as option to open it when I tap on map option. So, @nutpantz cannot open OSM Dashboard and I cannot open osmand :open_mouth:

**Comment by nutpantz** [comment]
![Screenshot_20220304-134728_Android System](https://user-images.githubusercontent.com/12160507/156846292-d2514825-df0c-4291-ba58-3012761ce0af.png)

That's what i see

**Comment by pstorch** [comment]
Right, same here. I have osmand installed, but I don't get osmand offered when sharing the track. But I remember, that worked in the past.

**Comment by pstorch** [comment]
Probably a strange idea, but the track name is truncated in the title bar. Can you try in landscape mode to push the map button?

**Comment by nutpantz** [comment]
If i freeze osmand then it's gone but is still only have the choice of open tracks or gps status

**Comment by nutpantz** [comment]
![Uploading Screenshot_20220304-135125_Android System.png…]()

**Comment by nutpantz** [comment]
![Screenshot_20220304-135125_Android System](https://user-images.githubusercontent.com/12160507/156846839-9fb43c72-6dab-4225-9af8-d17ed65f90ed.png)

**Comment by pstorch** [comment]
Ok, as I thought a strange idea. I guess you already tried a reboot? Sorry, I'm just guessing here at the moment.

**Comment by nutpantz** [comment]
Just tried, no help

**Comment by nutpantz** [comment]
If i freeze all my gps apps except open tracks i get (so the only gps app now installed is open tracks)
![Screenshot_20220304-135840_OpenTracks](https://user-images.githubusercontent.com/12160507/156847643-edeb7c0b-2ad6-461c-a7b8-11b6030c1fea.png)

**Comment by nutpantz** [comment]
Wide view
![Screenshot_20220304-140128_OpenTracks](https://user-images.githubusercontent.com/12160507/156847864-cb8c2015-150e-4686-9ff3-1539bb6aba4b.png)

**Comment by nutpantz** [comment]
I've got fresh download and install from fdroid for both apps.

**Comment by pstorch** [comment]
Ok, thanks for trying all those options. 
I'm still very puzzled. Osmand shouldn't show up there, because they don't listen on the OpenTracks dashboard API (as far as I know). But OSM Dashboard should. 
What is happening here?

**Comment by nutpantz** [comment]
Osmand not only shows up, but accepts the track and shows it in osmand

**Comment by nutpantz** [comment]
I would say somehow open tracks is not sending dashboards api but sending a gpx file that is acceptable by gps apps.
  
And i know I'm picking the map icon

**Comment by rgmf** [comment]
I remember I opened tracks with osmand. I'll try with other phones tomorrow... It the worst issue we can face against: it looks a completely random issue, it doesn't make any sense.

**Comment by pstorch** [comment]
One more try:
Can you open the track list in OpenTracks (the main screen), then long click on a track to select it, then the map icon should show up. Does this offer OSM Dashboard?

**Comment by nutpantz** [comment]
I've tried that way many times also. Still the same result. Same file name. Same everything

**Comment by pstorch** [comment]
I'm sorry, for today I'm out of ideas.

**Comment by nutpantz** [comment]
It is what it is. Thankfully there is snow on the ground still so it's not a huge pain

**Comment by nutpantz** [comment]
![Screenshot_20220304-142510_Android System](https://user-images.githubusercontent.com/12160507/156850116-ed0458ef-c98b-4ac0-b415-69441584e66b.png)

I only got this option on the first install of osm dashboard with internet, (or that's what my memory said) I'm sure i picked gpx then
I installed osm dashboard offline and i got the option again

**Comment by nutpantz** [comment]
So it's an issue with open tracks exporting the wrong format and no option to change it or a user knows it exists

**Comment by nutpantz** [comment]
I have no idea how to write this up as an issue for open track. Maybe someone else can help do that

**Comment by pstorch** [comment]
Ahhhhhh, now I got it.
Please go to the app info (long click on launcher icon), then Defaults/ Set as default, then clear defaults.

**Comment by nutpantz** [comment]
It's strange those defaults don't get cleared when the app is uninstalled. There needs to be a link in the app.

**Comment by pstorch** [comment]
When I once choose OpenTracks with gpx and choose always in the intent chooser, the I also get the same situation.

**Comment by pstorch** [comment]
> It's strange those defaults don't get cleared when the app is uninstalled. There needs to be a link in the app.

Looks like it's not resetted on uninstall. But I think Android remembers it internally and the app can't do anything about it.

**Comment by nutpantz** [comment]
No but the app can link to the setting (I'm sure I've seen that before) in case a new user (and I've been using open tracks for 2 years) makes a mistake when that option is asked. Or maybe remove the gpx and kml options for export so that the map icon only looks for osm dashboard
